### PR TITLE
Upgrade fxamacker/cbor to v2.5.0

### LINKED ISSUE #165: Signature.Sign doesn't check for malformed protected data containing extraneous bytes
### What is the areas you experience the issue in?

Go-COSE Library

### What is not working as expected?

`Signature.Sign` validates that the `protected` parameter contains a wellformed `bstr`, but it misses to detect byte arrays with more bytes than defined in the `bstr` prelude.

For example, `[]byte{0x40, 0xa1, 0x00, 0x00}` declares a `bstr` (major type 2) with 0 entries, but it contains 3 entries.

Notice that the validation happens inside `deterministicBinaryString`, which is also used in `Sign1Message.toBeSigned`, so it's possible that `Sign1Message` is also affected by this bug.

### What did you expect to happen?

`Signature.Sign` should return an error if `protected` contains extraneous bytes.

### How can we reproduce it?

```go
package main

import (
	"crypto/ecdsa"
	"crypto/elliptic"
	"crypto/rand"

	"github.com/veraison/go-cose"
)

func main() {
	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
	if err != nil {
		panic(err)
	}
	sig := cose.NewSignature()
	signer, err := cose.NewSigner(cose.AlgorithmES256, priv)
	if err != nil {
		panic(err)
	}
	err = sig.Sign(rand.Reader, signer, []byte{0x40, 0xa1, 0x00, 0x00}, []byte("Hello, world!"), nil)
	if err != nil {
		panic(err)
	}
}

```

### Describe your environment

Windows 11,  go1.21

### What is the version of your Go-COSE Library?

v1.1.0

### PR DESCRIPTION
**Pull request by qmuntal**
`fxamacker/cbor` release notes: https://github.com/fxamacker/cbor/releases/tag/v2.5.0.

Fixes #165

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

**Review by OR13 — APPROVED** [review]

**Review by OR13 — CHANGES_REQUESTED** [review]
On the call today, there was discussion on wanting "some number of additional unit tests" related to marshal and unmarshal.

This is vague, so any number of increases in unit tests would unblock my review.

I am requesting "additional unit tests" at the discretion of the maintainers.

**Comment by qmuntal** [comment]
> @qmuntal to evaluate what additional tests are required to build confidence to merge.

I've gone through the test suite of other COSE libraries, most notably https://github.com/google/coset, and I haven't found any outstanding missing case on our suite. IMO this is ready to merge.

**Review by OR13 — APPROVED** [review]

**Review by thomas-fossati — APPROVED** [review]

**Comment by SteveLasker** [comment]
Thanks, @qmuntal

**Pull request merged**
